src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.cs

```diff
@@ -1677,16 +1677,23 @@ private EventBuilder DefineEventNoLock(string name, EventAttributes attributes,
         #region Create Type
 
         [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
-        public TypeInfo? CreateTypeInfo()
+        public TypeInfo CreateTypeInfo()
         {
-            lock (SyncRoot)
-            {
-                return CreateTypeNoLock();
-            }
+            TypeInfo? typeInfo = CreateTypeInfoImpl();
+            Debug.Assert(typeInfo != null);
+            return typeInfo;
+        }
+
+        [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
+        public Type CreateType()
+        {
+            Type? type = CreateTypeInfoImpl();
+            Debug.Assert(type != null);
+            return type;
         }
 
         [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
-        public Type? CreateType()
+        internal TypeInfo? CreateTypeInfoImpl()
```

Comment:
The naming along the callchain looks a bit odd: `CreateType` -> `CreateTypeInfoImpl` -> `CreateTypeNoLock`

It may be nice to rename `TypeInfo? CreateTypeInfoImpl` to `RuntimeType? CreateRuntime`, and `TypeInfo? CreateTypeNoLock` to `RuntimeType? CreateRuntimeTypeNoLock`

Reply:
When I change from `TypeInfo?` to `RunetimeType?` The `this` at https://github.com/dotnet/runtime/blob/f4cb6ac85acbbbbcdc48893fea16d89fe898e669/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.cs#L1758  

I get a compile time error: 

`CS00029: Cannot implicitly convert type 'System.Reflection.Emit.TypeBuilder' to 'System.Runtime.Type'`

Reply:
Ok, I have not realized this detail. Let's keep it as is then.